core/src/test/java/org/elasticsearch/action/termvectors/GetTermVectorsIT.java

```diff
@@ -1195,6 +1198,30 @@ public void testFilterDocFreq() throws ExecutionException, InterruptedException,
         }
     }
 
+    public void testTook() throws Exception {
+        XContentBuilder mapping = jsonBuilder()
+            .startObject()
+                .startObject("type1")
+                    .startObject("properties")
+                        .startObject("field")
+                            .field("type", "text")
+                            .field("term_vector", "with_positions_offsets_payloads")
+                        .endObject()
+                    .endObject()
+                .endObject()
+            .endObject();
+        assertAcked(prepareCreate("test").addAlias(new Alias("alias")).addMapping("type1", mapping));
+
+        ensureGreen();
+
+        client().prepareIndex("test", "type1", "0").setSource("field", "foo bar").setRefresh(true).execute().get();
+        long start = System.nanoTime();
+        TermVectorsResponse response = client().termVectors(new TermVectorsRequest(indexOrAlias(), "type1", "0")).get();
+        assertThat(response, notNullValue());
+        assertThat(response.getTookInMillis(), greaterThanOrEqualTo(0L));
+        assertThat(response.getTookInMillis(), lessThanOrEqualTo(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start)));
```

Comment:
This test doesn't protect against insidious bugs. For example, if we modify:

```java
termVectorsResponse.setTookInMillis(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));
```

to

```java
termVectorsResponse.setTookInMillis((System.nanoTime() - startTime) / 10000000L);
```

then we've introduced a unit conversion bug but the test will not capture that because the test will still pass.